perovskite_solar_cell_database/schema_sections/perovskite.py

```diff
@@ -489,3 +572,76 @@ def normalize(self, archive, logger):
                 logger.warn('could not analyse chemical formula', exc_info=e)
             archive.results.material.elements = final_formula[1]
 
+        ions = []
+        a_ions_names = []
+        a_ions_coefficients = []
+        if self.composition_a_ions is not None:
+            a_ions_names = self.composition_a_ions.split("; ")
+        if self.composition_a_ions_coefficients is not None:
+            a_ions_coefficients = [float(c) for c in self.composition_a_ions_coefficients.split("; ")]
+
+        if len(a_ions_names) != 0 and len(a_ions_names) == len(a_ions_coefficients):
+           for i in range(len(a_ions_names)):
+               ion_a = Ion(name=a_ions_names[i], coefficients=a_ions_coefficients[i], ion_type='A')
+               ion_a.normalize(self, archive)
+               ions.append(ion_a)
+
+        b_ions_names = []
+        b_ions_coefficients = []
+        if self.composition_b_ions is not None:
+            b_ions_names = self.composition_b_ions.split("; ")
+        if self.composition_b_ions_coefficients is not None:
+            b_ions_coefficients = [float(c) for c in self.composition_b_ions_coefficients.split("; ")]
+
+        if len(b_ions_names) != 0 and len(b_ions_names) == len(b_ions_coefficients):
+           for i in range(len(b_ions_names)):
+                ion_b = Ion(name=b_ions_names[i], coefficients=b_ions_coefficients[i], ion_type='B')
+                ion_b.normalize(self, archive)
+                ions.append(ion_b)
+
+        c_ions_names = []
+        c_ions_coefficients = []
+        if self.composition_c_ions is not None:
+            c_ions_names = self.composition_c_ions.split("; ")
+        if self.composition_c_ions_coefficients is not None:
+            c_ions_coefficients = [float(c) for c in self.composition_c_ions_coefficients.split("; ")]
+        if len(c_ions_names) != 0 and len(c_ions_names) == len(c_ions_coefficients):
+           for i in range(len(c_ions_names)):
+               ion_c = Ion(name=c_ions_names[i], coefficients=c_ions_coefficients[i], ion_type='C')
+               ion_c.normalize(self, archive)
+               ions.append(ion_c)
+
+        self.ions = ions
+
+        from nomad.normalizing.topology import add_system_info, add_system
+        from nomad.datamodel.results import Relation
+        from nomad.normalizing.common import nomad_atoms_from_ase_atoms
+
+        topology = {}
+        # Add original system
+        atoms = nomad_atoms_from_ase_atoms(molecule('H2O'))
+        parent_system = System(
+            method='parser',
+            label='original',
```

Comment:
`label` here should not be original, should possibly be `absorber material`, or something like that.